test/clj/villagebook/fixtures.clj

```diff
@@ -1,16 +1,20 @@
 (ns villagebook.fixtures
-  (:require [villagebook.manage_migrations :as migrations]))
+  (:require [clojure.java.jdbc :as sql]
+            [villagebook.config :as config]
+            [villagebook.manage_migrations :as migrations]))
 
-(defn setup-test
+(defn setup-db
   []
   (migrations/migrate))
 
-(defn teardown-test
-  []
-  (migrations/rollback))
+(defn setup-once
+  [test]
+  (setup-db)
+  (test))
 
-(defn wrap-setup
-  [fun]
-  (setup-test)
-  (fun)
-  (teardown-test))
+(defn wrap-transaction
+  [test]
+  (sql/with-db-transaction [trn config/db-spec]
+    (sql/db-set-rollback-only! trn)
+    (binding [config/db-spec trn]
```

Comment:
You do not need to rebind `config/db-spec`

From the official docs for `sql/db-set-rollback-only!`
```
Marks the outermost transaction such that it will rollback rather than commit when complete
```
Thus `trn` will be rolled back. And iirc transactions in jdbc collapse to the top level txn. Thus any db-connection, txn you create from the db-spec this point onward will collapse into trn. 

Thus when trn rolls back everything underneath will also rollback.

Reply:
The rebinding is needed since we are not passing `trn` to the db functions. The jdbc calls in the db functions look like:
```clojure
(jdbc/query config/db-spec <query>)
```
These calls are not aware of the transaction since they call `config/db-spec` directly and make a new connection.